CMakeLists.txt

```diff
@@ -48,31 +51,43 @@ set(OVERPASS_SOURCES
 
 add_definitions("-std=c++11 -Wall -Wextra")
 
-add_library(overpass SHARED
+add_library(overpass
 	${OVERPASS_HEADERS}
 	${OVERPASS_SOURCES}
 )
 
+target_include_directories(overpass
+	PUBLIC
+		${PROJECT_SOURCE_DIR}/include
+		${PROJECT_BINARY_DIR}/include
+	PRIVATE
+		${Boost_INCLUDE_DIRS}
```

Comment:
Consider adding SYSTEM to both boost and libtins to avoid all the compiler warnings. (I didn't test it, you might see if it works first, if I get some time tonight I'll try it.)